courses/fundamentals_of_ada/quiz/README.md

```diff
@@ -0,0 +1,90 @@
+# Content of the directory
+
+This directory contain quizzes.
+
+Quizzes are **templates** for standalone Ada projects with multiple variants.
+
+Given a template, it is concretizable into several projects, one per
+choice. Those concrete projects will be *compile-run checked* by the tooling:
+those that compile and run without error will be considered correct answers.
+
+There are 3 kinds of errors that can be detected
+
+1. Project does not compile
+2. Project does compile but raises a runtime error
+3. Project does compile but raises a user error (`-gnata`)
+
+NB: For the case of user errors, the `-gnata` flag will need to be provided in the
+associated gpr file.
+
+Warning: Infinite loops or silent failures (return code is 0) are not detected by
+the tooling.
+
+# Usage
+
+The quizzes are meant to be used with the `contrib/quiz.py` script. This script
+takes an input file as argument and turns it into a Quiz slide in the training
+material's ReSTRuctured text idiom.
+
+This is done by instanciating copies of the quiz project directory, one for each
+choice (A, B, C...), extracting the associated question, trying a compile-run of
+the answer's code, and marking it as an answer in case it compiled and ran
+correctly.
+
+The output is saved as a *golden* .rst file and put into version control, so
+that the CI can double check the answer (using `contrib/quiz_update.py`), and
+validate that the result has not changed unexpectedly.
+
+This allows for reviewers to focus on the rst result, and for maintainers to
+focus on the Ada code. As a result, the quiz is reviewed in its included form,
+which is not compilable, and the Ada code is saved as well, allowing for
+unambiguous reproducibility.
+
+NB: Intermediate results could also be saved for similar reasons, but this is not
+the case today.
+
+# Template's format
+
+## adacut format
+
+The template is based on the `adacut` generic format, used by `contrib/adacut.py`.
+It provides *directives* that are to be written as
+
+```ada
+--$ directive cut [target]
+```
+
+* `directive` can be any of `begin`, `end`, `line`
+* `type` can be any of `cut`, `question`, `answer`
```

Comment:
I think this means "cut" on line 54 should be "<type>"

Reply:
Should be type yes